Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseAutonomous.java

```diff
@@ -2,24 +2,37 @@
 
 import com.qualcomm.robotcore.hardware.DcMotor;
 
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
 import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
 import org.firstinspires.ftc.robotcore.external.navigation.AxesOrder;
 import org.firstinspires.ftc.robotcore.external.navigation.AxesReference;
+import org.openftc.apriltag.AprilTagDetection;
+import org.openftc.easyopencv.OpenCvCamera;
+import org.openftc.easyopencv.OpenCvCameraFactory;
+import org.openftc.easyopencv.OpenCvCameraRotation;
 
-public abstract class BaseAutonomous extends BaseOpMode {
+import java.util.ArrayList;
+import java.util.Locale;
 
+public abstract class BaseAutonomous extends BaseOpMode {
     /**
      * this method will allow the robot to drive straight in a specified direction given a specified heading and distance
      * @param heading 360-degree direction robot should move (front is 0)
      * @param targetDistance distance robot should move in inches
      */
     public void driveInches(double heading, double targetDistance) {
+        // encoder values of the drive motors
+        int eFL, eFR, eBL, eBR;
+
         double startAngle = imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES).firstAngle;
         double turningPower;
 
         // looking at robot from the back, x is left/right and y is forwards/backwards
-        double xPosition;
-        double yPosition;
+        double xPosition, yPosition;
+
+        // power for any heading
+        double xPower = Math.cos(Math.toRadians(heading + 95)) * Constants.MAXIMUM_DRIVE_POWER;
```

Comment:
95 is a magic number here. Why is it being added to the heading?

Reply:
The unit circle is rotated 90 degrees relative to the front of the robot, it has been converted into a constant. The extra 5 degrees are a feedforward for our robot being unbalanced